### PR TITLE
SALTO-6284: Salesforce LWC deploy XML parsing bug-fix

### PR DESCRIPTION
**Pull request by tamtamirr**
Salesforce LWC deploy XML parsing bug-fix

---

There's a bug when a property has the attribute `default` with the value true. This is demonstrated in the UTs and pretty straightforward.

---
_Release Notes_: 
Salesfore Adapter:
- Fixed a bug where deploy of LightningWebComponent failed on `Attribute name "default" associated with an element type "property" must be followed by the ' = ' character.`

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/test/transformers/xml_transformer.test.ts [hunk 1]
```diff
@@ -281,9 +281,26 @@ describe('XML Transformer', () => {
...
+              {
+                name: 'testTrueProp',
+                default: 'true',
+              },
+              {
+                name: 'testFalseProp',
+                default: 'false',
+              },
+            ],
+          )
```
Comment (yelly) [inline packages/salesforce-adapter/test/transformers/xml_transformer.test.ts:300]: Maybe instead of this just add these properties to the mock? This will also help ensure we don't have this problem anywhere else.

File: packages/salesforce-adapter/src/transformers/xml_transformer.ts [hunk 2]
```diff
@@ -554,9 +560,11 @@ const cloneValuesWithAttributePrefixes = async (
     if (pathID && allAttributesPaths.has(pathID.getFullName())) {
       return XML_ATTRIBUTE_PREFIX + key
     }
+    if (pathID && trueValueAttributePaths.has(pathID.getFullName())) {
+      return `${XML_ATTRIBUTE_PREFIX}${key}="true"`
```
Comment (yelly) [inline packages/salesforce-adapter/src/transformers/xml_transformer.ts:564]: Please add a comment here that explains why we do this, it looks super weird.

File: packages/salesforce-adapter/test/transformers/xml_transformer.test.ts (hunk 1 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/test/transformers/xml_transformer.test.ts:300]: That was my previous commit. This broke the E2E deploy that used the same values which are mock and incomplete. I wouldn't complex it. We are testing edge-case for the XML parsing here

File: packages/salesforce-adapter/src/transformers/xml_transformer.ts (hunk 2 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/transformers/xml_transformer.ts:564]: I'll move the comment from L536 here

Comment (yelly) [reply packages/salesforce-adapter/src/transformers/xml_transformer.ts:564]: Please also specifically reference the issue with `fast-xml-parser` that causes us to do this. IIRC we have a comment about this in the other place we had this problem.

**Review by yelly — APPROVED** [review]
